tests/integration-tests/resources/file-cache-storage-cfn.yaml

```diff
@@ -0,0 +1,261 @@
+                    },
+                  ],
+                ).get("FileCache")
+              )
+
+            except fsx.exceptions.InternalServerError as e:
+              logger.info(f"Cannot create Fsx FileCache because of InternalServerError.")
+
+
+          def handler(event, context):
+            logger.info(f"Context: {context}")
+            logger.info(f"Event: {event}")
+            logger.info(f"Boto version: {boto3.__version__}")
+
+            file_cache_path =  event['ResourceProperties']['FileCachePath']
+            logger.info(f"file_cache_path: {file_cache_path}")
+
+            s3_bucket = 's3://{}'.format(event['ResourceProperties']['S3BucketName'])
+            logger.info(f"s3_bucket: {s3_bucket}")
+
+            subnet_id = event['ResourceProperties']['SubnetId']
+            logger.info(f"subnet_id: {subnet_id}")
+
+            securitygroup_id = event['ResourceProperties']['SecurityGroupId']
+            logger.info(f"securitygroup_id: {securitygroup_id}")
+
+
+            response_data = {}
+            reason = None
+            response_status = cfnresponse.SUCCESS
+
+            try:
+              if event['RequestType'] == 'Create':
+                file_cache_response = create_file_cache_id(file_cache_path, s3_bucket, subnet_id, securitygroup_id)
+                file_cache_id = file_cache_response.get("FileCacheId")
+                file_cache_arn= file_cache_response.get("ResourceARN")
+                response_data['Message'] = 'Resource creation successful!'
+                physical_resource_id = create_physical_resource_id()
+
+                logger.info(f"fsx_file_cache_id: {file_cache_id}")
+                # provide outputs
+                response_data['FileCacheId'] = file_cache_id
+                response_data['FileCacheARN'] = file_cache_arn
+              else:
+                  physical_resource_id = event['PhysicalResourceId']
+            except Exception as e:
+                response_status = cfnresponse.FAILED
+                reason = str(e)
+            cfnresponse.send(event, context, response_status, response_data, physical_resource_id, reason)
+
+
+
+  CreateFsxFileCache:
+    Type: Custom::CreateFsxFileCacheLambda
+    DependsOn:
+      - FsxFileCacheSecurityGroup
+    Properties:
+      ServiceToken: !GetAtt CreateFsxFileCacheLambda.Arn
+      FileCachePath: !Ref FileCachePath
+      S3BucketName: !Ref S3BucketName
+      SubnetId: !Ref SubnetId
+      SecurityGroupId: !Ref FsxFileCacheSecurityGroup
+
+  FsxFileCacheLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonFSxFullAccess
+        - arn:aws:iam::aws:policy/AmazonFSxConsoleFullAccess
```

Comment:
Why is `AmazonFSxConsoleFullAccess ` required?

Reply:
My stack was failing the creation of Fsx FC due to insufficient access. In order to unblock myself i added these, will remove them.